frontend/src/components/UploadPipelineDialog.tsx

```diff
@@ -71,54 +79,87 @@ class UploadPipelineDialog extends React.Component<UploadPipelineDialogProps, Up
+            <FormControlLabel id='uploadLocalFileBtn' label='Upload a file' checked={importMethod === ImportMethod.LOCAL}
+              control={<Radio color='primary' />} onChange={() => this.setState({ importMethod: ImportMethod.LOCAL })} />
+            <FormControlLabel id='uploadFromUrlBtn' label='Import by URL' checked={importMethod === ImportMethod.URL}
+              control={<Radio color='primary' />} onChange={() => this.setState({ importMethod: ImportMethod.URL })} />
           </div>
-          <Input onChange={this.handleChange('fileName')} value={fileName} required={true} label='File'
-            InputProps={{
-              endAdornment: (
-                <InputAdornment position='end'>
-                  <Button color='secondary' onClick={() => this._dropzoneRef.current!.open()}
-                    style={{ padding: '3px 5px', margin: 0 }}>
-                    Choose file
-               </Button>
-                </InputAdornment>
-              ),
-              readOnly: true,
-            }} />
+
+          {importMethod === ImportMethod.LOCAL && (
+            <React.Fragment>
+              <Dropzone id='dropZone' disableClick={true}
+                onDrop={this._onDrop.bind(this)} onDragEnter={this._onDropzoneDragEnter.bind(this)}
+                onDragLeave={this._onDropzoneDragLeave.bind(this)} style={{ position: 'relative' }}
+                ref={this._dropzoneRef} inputProps={{ tabIndex: -1 }}>
+
+                {dropzoneActive && (
+                  <div className={css.dropOverlay}>Drop files..</div>
+                )}
+
+              <div className={padding(10, 'b')}>
+                Choose a pipeline package file from your computer, and give the pipeline a unique name.
+                <br />
+                You can also drag and drop the file here.
+              </div>
+              <Input onChange={this.handleChange('fileName')} value={fileName} required={true}
+                label='File'
+                InputProps={{
+                  endAdornment: (
+                    <InputAdornment position='end'>
+                      <Button color='secondary' onClick={() => this._dropzoneRef.current!.open()}
+                        style={{ padding: '3px 5px', margin: 0 }}>
+                        Choose file
+                      </Button>
+                    </InputAdornment>
+                  ),
+                  readOnly: true,
+                }} />
+              </Dropzone>
+            </React.Fragment>
+          )}
+
+          {importMethod === ImportMethod.URL && (
+            <React.Fragment>
+              <div className={padding(10, 'b')}>
+                URL must be publicly accessible.
+              </div>
+              <Input onChange={this.handleChange('fileUrl')} value={fileUrl} required={true} label='URL' />
+            </React.Fragment>
+          )}
 
           <Input id='uploadFileName' label='Pipeline name'
```

Comment:
We should remove the `upload` parts of the id and the state field here since they're also used in url import path.

Reply:
Hm, I don't disagree, but the entire component is based around that name (`UploadPipelineDialog`). I'm up for renaming it in a follow-up, but what's a better verb? `Add`?

Reply:
Good point, probably good to handle that separately if we see the need.

/lgtm